unime/src/routes/+layout.svelte

```diff
@@ -57,46 +59,121 @@
+  async function loadFerrisProfile() {
+    dispatch({ type: '[DEV] Load DEV profile', payload: { profile: 'Ferris', execute_steps: null } }).then(() => {
+      // Reload page
+      setTimeout(async () => {
+        await goto('/');
+        await goto('/me');
+      }, 500);
+    });
+  }
+
+  async function loadDragonProfile(steps: ProfileSteps) {
+    await dispatch({
+      type: '[DEV] Load DEV profile',
+      payload: {
+        profile: 'Dragon',
+        execute_steps: steps,
+      },
+    });
+
+    showDragonProfileSteps = false;
+
+    if (steps == 'CompleteFlow') {
+      // Reload page
+      setTimeout(async () => {
+        await goto('/');
+        await goto('/me');
+      }, 500);
+    }
+  }
 </script>
 
 <main class="absolute h-screen">
   <!-- Dev Mode: Navbar -->
-  {#if $state?.dev_mode_enabled}
-    {#if showDevMode}
+  {#if $state?.dev_mode !== 'Off'}
+    {#if expandDevMenu}
       <div
-        class="hide-scrollbar fixed z-20 flex w-full space-x-4 overflow-x-auto bg-gradient-to-r from-red-200 to-red-300 p-4 shadow-md"
+        class="hide-scrollbar fixed z-20 flex w-full content-center overflow-x-auto bg-gradient-to-r from-red-200 to-red-300 p-4 pt-8 shadow-md"
         in:fly={{ y: -64, opacity: 1 }}
         out:fly={{ y: -64, opacity: 1 }}
       >
-        <button
-          class="rounded-full bg-red-300 px-4 py-1 text-sm font-medium text-red-700 hover:outline-none hover:ring-2 hover:ring-red-700 hover:ring-opacity-60"
-          on:click={() => history.back()}
-        >
-          <ArrowLeft class="h-6 w-6" />
-        </button>
-        <button
-          class="flex-shrink-0 rounded-full bg-red-300 px-4 py-2 text-sm font-medium text-red-700 hover:outline-none hover:ring-2 hover:ring-red-700 hover:ring-opacity-60"
-          on:click={() => dispatch({ type: '[App] Reset' })}><Trash class="h-6 w-6" /></button
-        >
-        <button
-          class="flex-shrink-0 rounded-full bg-red-300 px-4 py-2 text-sm font-medium text-red-700 hover:outline-none hover:ring-2 hover:ring-red-700 hover:ring-opacity-60"
-          on:click={() => dispatch({ type: '[DEV] Load profile' })}
-        >
-          <span class="text-[18px]/[18px]">🦀</span>
-        </button>
-
-        <!-- Debug messages -->
-        <button
-          class="flex-shrink-0 rounded-full bg-red-300 px-4 py-2 text-sm font-medium text-red-700 hover:outline-none hover:ring-2 hover:ring-red-700 hover:ring-opacity-60"
-          on:click={() => (showDebugMessages = !showDebugMessages)}><Warning class="h-6 w-6" /></button
-        >
+        {#each devButtons as btn}
+          <div class="mr-4 flex flex-col content-start justify-center">
+            <button
+              class="m-0 flex h-10 w-10 content-center rounded-full bg-red-300 p-0 text-sm
+                font-medium text-red-700 hover:outline-none hover:ring-2 hover:ring-red-700 hover:ring-opacity-60"
+              on:click={btn.onClick}
+            >
+              {#if btn.stringIcon}
```

Comment:
What if both `stringIcon` and `svelteIcon` are undefined? Wouldn't it make more sense to just define one `icon` that is not optional and can have both types?

Reply:
No because the checing of undefined in the if statement is way more clean like this

Reply:
> What if both `stringIcon` and `svelteIcon` are undefined? Wouldn't it make more sense to just define one `icon` that is not optional and can have both types?

You say the checking of undefined it way more clean, but you're only doing it for `stringIcon` and in the other case you're just `<svelte:component this={btn.svelteIcon} />` without checking for undefined.